commit 89376da80b2250b82d256ea85ec349ce29fe5b51
Author: Stephen Gallagher <sgallagh@redhat.com>
Date:   Thu May 5 10:44:24 2016 -0400

    GPO: Add "unity" to ad_gpo_map_interactive
    
    Ubuntu systems use "unity" as their screen-locker. Without this in the
    defaults, people often get locked out of their machines when the screen
    locks.
    
    Resolves:
    https://bugs.launchpad.net/ubuntu/+source/sssd/+bug/1578415
    
    Reviewed-by: Lukáš Slebodník <lslebodn@redhat.com>

diff --git a/src/man/sssd-ad.5.xml b/src/man/sssd-ad.5.xml
index dc17a2f..265409e 100644
--- a/src/man/sssd-ad.5.xml
+++ b/src/man/sssd-ad.5.xml
@@ -439,6 +439,11 @@ ad_gpo_map_interactive = +my_pam_service, -login
                                 </listitem>
                                 <listitem>
                                     <para>
+                                        unity
+                                    </para>
+                                </listitem>
+                                <listitem>
+                                    <para>
                                         xdm
                                     </para>
                                 </listitem>
diff --git a/src/providers/ad/ad_gpo.c b/src/providers/ad/ad_gpo.c
index 3029ffe..22ac803 100644
--- a/src/providers/ad/ad_gpo.c
+++ b/src/providers/ad/ad_gpo.c
@@ -188,6 +188,7 @@ int ad_gpo_process_cse_recv(struct tevent_req *req);
 #define GPO_LIGHTDM "lightdm"
 #define GPO_LXDM "lxdm"
 #define GPO_SDDM "sddm"
+#define GPO_UNITY "unity"
 #define GPO_XDM "xdm"
 #define GPO_SSHD "sshd"
 #define GPO_FTP "ftp"
@@ -209,7 +210,7 @@ struct gpo_map_option_entry {
 const char *gpo_map_interactive_defaults[] =
     {GPO_LOGIN, GPO_SU, GPO_SU_L,
      GPO_GDM_FINGERPRINT, GPO_GDM_PASSWORD, GPO_GDM_SMARTCARD, GPO_KDM,
-     GPO_LIGHTDM, GPO_LXDM, GPO_SDDM, GPO_XDM, NULL};
+     GPO_LIGHTDM, GPO_LXDM, GPO_SDDM, GPO_UNITY, GPO_XDM, NULL};
 const char *gpo_map_remote_interactive_defaults[] = {GPO_SSHD, GPO_COCKPIT,
                                                      NULL};
 const char *gpo_map_network_defaults[] = {GPO_FTP, GPO_SAMBA, NULL};
